js/index.js

```diff
@@ -12,61 +12,44 @@
 
 import {Platform} from 'react-native';
 import {RNCNetInfo, NetInfoEventEmitter} from './nativeInterface';
+import type {
+  NetInfoData as _NetInfoData,
+  NetInfoType as _NetInfoType,
+  NetInfoEffectiveType as _NetInfoEffectiveType,
+} from './types';
+
+export type NetInfoData = _NetInfoData;
+export type NetInfoType = _NetInfoType;
+export type NetInfoEffectiveType = _NetInfoEffectiveType;
 
 const DEVICE_CONNECTIVITY_EVENT = 'networkStatusDidChange';
+const CHANGE_EVENT_NAME = 'connectionChange';
 
 type ChangeEventName = 'connectionChange';
 
-export type ConnectionType =
-  // iOS & Android
-  | 'none'
-  | 'cellular'
-  | 'unknown'
-  | 'wifi'
-  // Android only
-  | 'bluetooth'
-  | 'ethernet'
-  | 'wimax';
-
-export type EffectiveConnectionType = 'unknown' | '2g' | '3g' | '4g';
-
-export type NetInfoData = {
-  type: ConnectionType,
-  effectiveType: EffectiveConnectionType,
-};
-
 type ChangeHandler = (data: NetInfoData) => void;
-
 type IsConnectedHandler = (isConnected: boolean) => void;
+// Ideally we would use the EmitterSubscription from react-native, but it is not publy exported
```

Comment:
typo in a comment, it's there but I feel almost silly mentioning it

Reply:
Never too silly to mention!